Source/WebCore/platform/graphics/ca/TileController.cpp

```diff
@@ -125,12 +233,18 @@ void TileController::tileCacheLayerBoundsChanged()
 
 void TileController::setNeedsDisplay()
 {
+    if (m_asyncTileConfigurationChangeData)
```

Comment:
This might fall into the category of 'too annoying at this point', but could it make sense to have the old tile grid be the one that's stored in a new variable, and the 'pending' grid be the primary tileGrid?

It seems like most operations want to operate on the pending one, we just want to keep the old one around so that it doesn't remove the tiles.